roles/satellite-clone/tasks/main.yml

```diff
@@ -218,6 +224,11 @@
 - block:
   # restore backup data
   - include_tasks: restore.yml
+    when: satellite_version in ["6.2", "6.3", "6.4"]
+
+  - name: restore using foreman-maintain
+    command: foreman-maintain restore --assumeyes {{ backup_dir }}
+    when: satellite_version in ["6.5"]
```

Comment:
I was able to restore a backup from a machine which had a password other than changeme. Not sure where that password is coming from though; I don't have a foreman-maintain-hammer.yml on the satellite I cloned from.

Reply:
I think its something worth asking about, I believe foreman-maintain will ask for your hammer credentials on upgrades and store them in `foreman-maintain-hammer.yml`. This may be more of a relevant thing for upgraded systems. Though, I don't think `foreman-maintain restore` actually uses hammer so the credentials probably aren't needed here. It looks like it will [use the one from satellite-answers](https://github.com/theforeman/foreman_maintain/commit/612a97632584636f9a8ff7927b46916972ddf61f) even if it does, so it seems like less of an issue than before.

Maybe @mbacovsky or @kgaikwad could clarify, will `foreman-maintain restore` ever need hammer credentials stored in `foreman-maintain-hammer.yml`?

Reply:
IIRC, in case of `foreman-maintain restore` there is no any step which uses hammer so `foreman-maintain restore` don't need to have it.  

With the latest hammer credential changes in the foreman-maintain,
- Initially, it uses credential from hammer config files to connect 
     - If that works then it uses same credential & doesn't store it into file `foreman-maintain-hammer.yml`.
     - If fails then tries to connect using installer's admin credential
          - If works then it saves credential to `foreman-maintain-hammer.yml` 
          - If fails then asks correct credential to user & store it in the same file on successful connection.
          - Then after foreman-maintain checks hammer connection using `foreman-maintain-hammer.yml` & repeat above steps on failure.

@mbacovsky, please correct me if anything is missing.

Reply:
That would explain why the restore step is working with a modified password in the backup. Thanks for the clarification!

Reply:
Thanks @kgaikwad!

@jturel seems like what you have will work fine for our purposes :+1: